File: 0283. Move Zeroes/move-zeroes.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int counter = 0;
        vector <int> temp (nums.size(), 0);
        for(int i=0;i<nums.size();i++){
            if(nums[i]!=0){
                temp[counter++] = nums[i];
            }
        }
        nums = temp;
    }
};

/*
Runtime: 33 ms, faster than 66.29% of C++ online submissions for Move Zeroes.
Memory Usage: 19.7 MB, less than 6.32% of C++ online submissions for Move Zeroes.
*/

File: 0283. Move Zeroes/move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let counter = 0;
    for(var i=0;i<nums.length;i++){
        if(nums[i]!==0){
            nums[counter++] = nums[i];
        }
    }
    while(counter<nums.length){
        nums[counter++] = 0;
    }
};

/*
Runtime: 165 ms
*/

// OR


/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */

const swap = function(nums, x,y){
	var tmp = nums[x];
	nums[x] = nums[y];
	nums[y] = tmp;
}

var moveZeroes = function(nums) {
    let counter = 0;
    for(var i=0;i<nums.length;i++){
        if(nums[i]!==0){
            swap(nums, counter++,i);
        }
    }
};

/*
Runtime: 91 ms
*/